### PR TITLE
Server can delete recipient from database

### PR DESCRIPTION
**Pull request by blackenjoy27**
# Description

The database has restriction for deleting recipient out of the table, our code has change the restriction for deletion by modifying the migration files for the database and adding new function for the recipient model file.

Summary of your changes

Changed the restriction for delete action specifically on the recipient table.
Addding delete function for recipient model file.

## What work was done?

Created a function call deleteRecipient
Change the parameter for .onDelete from "RESTRICT" to "CASCADE"

## Why was this work done?

The server is having trouble deleting recipient because not only the delete function that it use is not working for the router but also it violate the foreign key constraint.

## Type of change

Bug Fix

## Change Status

- Completed, ready to review and merge?
- In progress?
- etc.

## Checklist

- [x] Endpoint tested with Postman, or Unit test.
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] There are no merge conflicts

### CODE REVIEW

File: __tests__/routes/recipient.test.js [hunk 1]
```diff
@@ -17,6 +17,7 @@ const testUser1 = {
   zip_code: "08259-2106",
   veteran_status: false,
   household_size: 8
+  // ""
```
Comment (ahmedseragcodes) [inline __tests__/routes/recipient.test.js:20]: Not quite sure what this is? But looks like a comment you no longer need, maybe remove that.

File: data/migrations/014-email_addresses.js [hunk 2]
```diff
@@ -8,7 +8,7 @@ exports.up = function (knex) {
       .references('recipient_id')
       .inTable('recipients')
       .onUpdate('CASCADE')
-      .onDelete('RESTRICT');
+      .onDelete('CASCADE');
```
Comment (ahmedseragcodes) [inline data/migrations/014-email_addresses.js:11]: Might want to confirm that we want a delete to cascade throughout the app upon deleting a recipient. Possibly with stakeholder?

File: data/migrations/015-phone_numbers.js [hunk 3]
```diff
@@ -8,7 +8,7 @@ exports.up = function (knex) {
       .references('recipient_id')
       .inTable('recipients')
       .onUpdate('CASCADE')
-      .onDelete('RESTRICT');
+      .onDelete('CASCADE');
```
Comment (ahmedseragcodes) [inline data/migrations/015-phone_numbers.js:11]: Might want to confirm that we want a delete to cascade throughout the app upon deleting a recipient. Possibly with stakeholder?

**Review by ahmedseragcodes — APPROVED** [review]
Looks good based on what this is fixing. I would just maybe confirm the CASCADE for the deletion of email/phone with stakeholder? But easy to reverse if that's not wanted.

File: data/migrations/014-email_addresses.js (hunk 2 above)
Comment (blackenjoy27) [reply data/migrations/014-email_addresses.js:11]: Yes, I think this is the only way for database to delete data.

**Review by chance10113 — APPROVED** [review]
As long as we are allowing built-in functionality to remove recipients (I can't imagine why not, this is CRUD after all), and these migration tables correspond to the recipient, then this looks good to me.

**Review by blackenjoy27 — COMMENTED** [review]
I think this is the correct way to allow the database remove data.

**Pull request merged**